bot/utils/decorators.py

```diff
@@ -199,23 +199,38 @@ def predicate(ctx: Context) -> bool:
         kwargs = default_kwargs.copy()
         allow_dms = False
 
-        # Update kwargs based on override
-        if hasattr(ctx.command.callback, "override"):
+        # determine which command's overrides we will use
+        # as we have groups, we want to ensure that group commands inherit from the parent
+        overridden_command: Union[commands.Command, commands.Group] = None
```

Comment:
Huh, shouldn't the type be `Union[commands.Command, commands.Group, None]`?

---

Also by the looks of it you don't use any `commands.Group`-specific attributes and it's already a subclass of `commands.Command` so you should be able to do `Optional[commands.Command]` unless I am missing something.